zkvm/src/prover.rs

```diff
@@ -57,6 +57,16 @@ impl<'a, 'b> Delegate<r1cs::Prover<'a, 'b>> for Prover<'a, 'b> {
         Ok(run.program.pop_front())
     }
 
+    fn new_run(&self, program: Data) -> Result<Self::RunType, VMError> {
+        let instructions = match program {
```

Comment:
maybe factor this out to `Data::to_program` converter like we do with `to_predicate`, `to_commitment` etc?